From e56f94e5b4a6cf5530d2830fc95437fc57f46ca5 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 13 Feb 2018 09:33:42 -0800
Subject: [PATCH 0271/1851] drm: xlnx: xlnx_drv: Move xlnx bridge helper
 initialization

commit 6a8afdb43734ca7ecae933b1887773b01e97ff51 from
https://github.com/Xilinx/linux-xlnx.git

Since the xlnx_drv becomes more like a library, xlnx bridge can
be initilized when the module is loaded. This will allow
any client to defer the probe if helper or bridge provider are
not available.

The change will have to submitted along with Xilinx bridge
patch set.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_drv.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/xlnx/xlnx_drv.c b/drivers/gpu/drm/xlnx/xlnx_drv.c
index 81cb90b3fa81..83b97aa38f08 100644
--- a/drivers/gpu/drm/xlnx/xlnx_drv.c
+++ b/drivers/gpu/drm/xlnx/xlnx_drv.c
@@ -373,7 +373,6 @@ static int xlnx_compare_of(struct device *dev, void *data)
 
 static int xlnx_platform_probe(struct platform_device *pdev)
 {
-	xlnx_bridge_helper_init();
 	return xlnx_of_component_probe(&pdev->dev, xlnx_compare_of,
 				       &xlnx_master_ops);
 }
@@ -381,7 +380,6 @@ static int xlnx_platform_probe(struct platform_device *pdev)
 static int xlnx_platform_remove(struct platform_device *pdev)
 {
 	component_master_del(&pdev->dev, &xlnx_master_ops);
-	xlnx_bridge_helper_fini();
 	return 0;
 }
 
@@ -493,6 +491,7 @@ EXPORT_SYMBOL_GPL(xlnx_drm_pipeline_exit);
 
 static int __init xlnx_drm_drv_init(void)
 {
+	xlnx_bridge_helper_init();
 	platform_driver_register(&xlnx_driver);
 	return 0;
 }
@@ -500,6 +499,7 @@ static int __init xlnx_drm_drv_init(void)
 static void __exit xlnx_drm_drv_exit(void)
 {
 	platform_driver_unregister(&xlnx_driver);
+	xlnx_bridge_helper_fini();
 }
 
 module_init(xlnx_drm_drv_init);
-- 
2.31.1

